specs/data_structures.md

```diff
@@ -475,7 +477,28 @@ enum VoteType : uint8_t {
 
 # State
 
-TODO validator set repr
+The state of the LazyLedger chain contains only account balances and the validator set (which is really just extra metadata on top of the account balances).
+
+Two [Sparse Merkle Trees](#sparse-merkle-tree) are maintained: one of [accounts](#account) and one of [validators](#validator). The state root is computed as the [hash](#hashdigest) of concatenation of the account tree root and the validator tree root.
+
+## Account
+
+| name                 | type                | description                                                                              |
+| -------------------- | ------------------- | ---------------------------------------------------------------------------------------- |
+| `balance`            | `uint64`            | Coin balance.                                                                            |
+| `isDelegating`       | `bool`              | Whether this account is delegating its stake or not.                                     |
+| `delegatedValidator` | [Address](#address) | _Optional._ The validator this is account is delegating to.                              |
```

Comment:
Why can't accounts delegate to several validators?

Reply:
Because that would add complexity (you'd have to hold a variable number of validators and the number of coins each of them is delegated), and if someone wants to delegate their coins to two validators they can just...split up their coins into two accounts.

Reply:
This scheme (which isn't obvious from the data structures, but will be written in the consensus rules) is: "all the coins in an account are either delegated or not; if they're delegated they must be delegated to a single validator."

Reply:
From a user perspective holding coins and a UX perspective, splitting up coins into separate accounts whenever you want to delegate to more than one validator also adds complexity but one the human (not on the software).

Note that in the few delegations based PoS systems  that launched, you can usually delegate to one or more validators. This is also true in Cosmos / the cosmos-sdk. Most users I know delegate to several validators to keep the network decentralized (and to hedge their risks).

Reply:
Hmm. I don't see why this can't be abstracted away at the application layer, rather than having to be pushed to the user layer. Moving a portion of your coins to a new address then delegating them (as well as the reverse) can be done just as seamlessly in a wallet---one button press. Users don't even have to know it's happening! It's basically UTXO management; most Bitcoin wallets don't present to lay users each individual UTXO, just their total balance.

Allowing an arbitrary number of delegations per account means fraud proofs are more complex and expensive.

That being said, given that we're using the accounts data model for now, one downside of essentially emulating UTXOs for delegating stake is that the nonce of empty accounts still needs to be kept around forever (which reminds me, a nonce field needs to be added).

Reply:
I feel that it would actually be more intuitive from a UX perspective to require users to split their coins into different accounts to delegate to different validators. It's the logical purpose of accounts - the same way you might have multiple savings accounts for different purposes.

It seems more complicated from a UX perspective to have users enter each validator they want to delegate to and the amount in some kind of table.

Reply:
> It's the logical purpose of accounts - the same way you might have multiple savings accounts for different purposes.

But here you'd have multiple accounts for one purpose: namely delegate to validators. Having a different account per validator would be like opening a broker account per same type of investment.